script.js

```diff
@@ -1,4 +1,190 @@
+  console.log(`user win ${USER_WIN}`);
+  return gameResult;
+};
+
+var interpretResult = function (gameResult) {
+  var choiceMsg = `The program chose ${progChoice} ${createEmoji(progChoice)}.<br>
+  You chose ${userChoice} ${createEmoji(userChoice)}.`;
+  if (gameResult == DRAW) {
+    numGames += 1;
+    numDraws += 1;
+    choiceMsg += `<br><br>Oops, it's a draw!`;
+  } else if (gameResult == COM_WIN) {
+    numGames += 1;
+    numComWins += 1;
+    choiceMsg += `<br><br>Shucks, you lost!`;
+  } else if (gameResult == USER_WIN) {
+    numGames += 1;
+    numUserWins += 1;
+    choiceMsg += `<br><br>Beginner's luck - you somehow managed to win!`;
+  } else if (gameResult == NO_WINNER_YET) {
+    choiceMsg += `<br><br>No one has won a round yet - please continue playing. Muk...jji...ppa!`;
+  } else if (gameResult == PENDING_DRAW) {
+    choiceMsg += `<br><br>The latest winner is: ${latestWinner}. If the next round is a draw, then ${latestWinner} will be the winner! Muk...jji...ppa!`;
+  }
+  var userWinPct = (numUserWins / numGames) * 100;
+  var numWinsMsg = `So far ${userName}, out of the ${numGames} game(s) played, you have won ${numUserWins} game(s), the computer has won ${numComWins} game(s) and there have also been ${numDraws} draw(s). You therefore have a winning percentage of ${userWinPct.toFixed(
+    2
+  )}%.`;
+  if (userWinPct >= WINNING_PCT_THRESHOLD) {
+    numWinsMsg += ` Well done so far! 👏`;
+  } else {
+    numWinsMsg += ` Disappointing, try harder! 🤦`;
+  }
+  var finalMsg = `${choiceMsg}<br><br>
+  ${numWinsMsg}`;
+  return finalMsg;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue;
+  if (gameMode == PENDING_USER_NAME) {
+    userName = input;
+    gameMode = PENDING_GAME_MODE;
+    myOutputValue = `Hello, ${input}. Type "${NORMAL}", "${REVERSE}", "${COM_VS_COM}" or "${KOREAN}" to select the game mode.`;
+  } else if (gameMode == PENDING_GAME_MODE) {
+    if (input == REVERSE) {
+      gameMode = REVERSE;
+      myOutputValue = `You have selected ${REVERSE} mode! Type "${SCISSORS}", "${PAPER}" or "${STONE}" to play.`;
+    } else if (input == NORMAL) {
+      gameMode = NORMAL;
+      myOutputValue = `You have selected ${NORMAL} mode! Type "${SCISSORS}", "${PAPER}" or "${STONE}" to play.`;
+    } else if (input == COM_VS_COM) {
+      gameMode = COM_VS_COM;
+      myOutputValue = `You have selected ${COM_VS_COM} mode! Click on "Submit" to play.`;
+    } else if (input == KOREAN) {
+      gameMode = KOREAN;
+      myOutputValue = `You have selected ${KOREAN} mode! Type "${SCISSORS}", "${PAPER}" or "${STONE}" to play.`;
+    } else {
+      myOutputValue = `Please type "${NORMAL}", "${REVERSE}", "${COM_VS_COM}" or "${KOREAN}" to select the game mode.`;
+    }
+  } else if (gameMode == NORMAL) {
+    if (input != SCISSORS && input != PAPER && input != STONE) {
+      myOutputValue = `Please type "${SCISSORS}", "${PAPER}" or "${STONE}" to play normal mode.`;
```

Comment:
Consider abstracting the generating of the outputvalue message to a separate function.